### ISSUE TITLE
Question about the Tutorial

### ISSUE THREAD

**Issue by scarlehoff** [microsoftgraph/msgraph-training-nodeexpressapp#50]
Wasn't sure where to leave this question, so I'll do it here.

I was following the tutorial [here](https://docs.microsoft.com/en-us/graph/tutorials/node?tutorial-step=3) and I find the constant replacements of `signInComplete` a bit confusing. In that page it gets rewritten 3 times. I'm not sure whether there is a reason for that*

Don't want this to sound mean or anything :cold_sweat:, I genuinely afraid there is something I'm not understanding

**Comment by jasonjoh** [comment]
Hi @scarlehoff, thanks for your question and your interest in Microsoft Graph! I understand that the re-writing can be confusing. The intention was to build bit-by-bit to let you see how the authentication and token acquisition process works behind the scenes.

- First iteration: bare-bones required for passport auth to complete.
- Second iteration: add using Microsoft Graph to pull the user's profile information during the auth flow and add their email address to the stored profile.
- Third iteration: build simple-oauth2 token object and store that and user profile in user storage.

Let me know if that answers your question.

**Comment by scarlehoff** [comment]
Oh, ok, I guess the constant reference to "replace" made me loss the "why" (specially in the second iteration).

Thanks.
If I had to give a suggestion, I would reword a bit at least the third iteration, since although it says replace it just ads a few lines (I had to open both side by side, try to see the differences and then got confused when I didn't see it and came to open the issue :grimacing: )

**Comment by jasonjoh** [comment]
That's good feedback! This one's on my backlog to refresh soon so I'll take a crack at making that bit clearer.
